src/api/v1/community_businesses/volunteers/__tests__/projects.test.integration.ts

```diff
@@ -0,0 +1,253 @@
+import { CommunityBusinesses, Organisation } from '../../../../../models';
+
+
+describe('API /community-businesses/{id}/volunteers', () => {
+  let server: Hapi.Server;
+  let knex: Knex;
+  let organisation: Organisation;
+  const config = getConfig(process.env.NODE_ENV);
+
+  beforeAll(async () => {
+    server = await init(config);
+    knex = server.app.knex;
+    organisation = await CommunityBusinesses.getOne(knex, { where: { id: 2 } });
+  });
+
+  afterAll(async () => {
+    await server.shutdown(true);
+  });
+
+  describe('GET /community-businesses/me/volunteers/projects', () => {
+    test('can fetch own projects', async () => {
+      const res = await server.inject({
+        method: 'GET',
+        url: '/v1/community-businesses/me/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:read'],
+          organisation,
+        },
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.result).toEqual({
+        result: [
+          { name: 'Party', organisationId: organisation.id },
+          { name: 'Take over the world', organisationId: organisation.id },
+        ].map(expect.objectContaining),
+      });
+    });
+
+    test('cannot fetch other orgs projects', async () => {
+      const res = await server.inject({
+        method: 'GET',
+        url: '/v1/community-businesses/2/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:read'],
+          organisation,
+        },
+      });
+
+      expect(res.statusCode).toBe(404);
+    });
+  });
+
+  describe('POST /community-businesses/me/volunteers/projects', () => {
+    test('can create new project for own org', async () => {
+      const res = await server.inject({
+        method: 'POST',
+        url: '/v1/community-businesses/me/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:write'],
+          organisation,
+        },
+        payload: {
+          name: 'new project',
+        },
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect(res.result).toEqual({
+        result: expect.objectContaining({
+          organisationId: organisation.id,
+          name: 'new project',
+        }),
+      });
+
+      const res2 = await server.inject({
+        method: 'GET',
+        url: '/v1/community-businesses/me/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:read'],
+          organisation,
+        },
+      });
+
+      expect(res2.statusCode).toBe(200);
+      expect((<any> res2.result).result).toHaveLength(3);
+    });
+
+    test('cannot create new project for other org', async () => {
+      const res = await server.inject({
+        method: 'POST',
+        url: '/v1/community-businesses/2/volunteers/projects',
+        credentials: {
+          scope: ['volunteer_logs-own:write'],
+          organisation,
+        },
+      });
+
+      expect(res.statusCode).toBe(404);
```

Comment:
why `404` not `401`?

Reply:
Because the general `{id}` route has not been implemented.